docs/docs/classic-ml/deep-learning/tensorflow/guide/index.mdx

```diff
@@ -0,0 +1,676 @@
+</details>
+
+### Customizing MLflow Logging
+
+You can create your own callback by subclassing `keras.callbacks.Callback` to implement custom logging logic:
+
+```python
+from tensorflow import keras
+import math
+import mlflow
+
+
+class CustomMlflowCallback(keras.callbacks.Callback):
+    def on_epoch_begin(self, epoch, logs=None):
+        mlflow.log_metric("current_epoch", epoch)
+
+    def on_epoch_end(self, epoch, logs=None):
+        logs = logs or {}
+        # Log metrics in log scale
+        for k, v in logs.items():
+            if v > 0:  # Avoid log(0) or log(negative)
+                mlflow.log_metric(f"log_{k}", math.log(v), step=epoch)
+            mlflow.log_metric(k, v, step=epoch)
+
+    def on_train_end(self, logs=None):
+        # Log final model weights statistics
+        weights = self.model.get_weights()
+        mlflow.log_metric("total_parameters", sum(w.size for w in weights))
+        mlflow.log_metric(
+            "average_weight",
+            sum(w.sum() for w in weights) / sum(w.size for w in weights),
+        )
+```
+
+<details>
+  <summary>Keras Callback Lifecycle Hooks</summary>
+
+  Keras callbacks provide various hooks into the training process:
+
+  - **Training Setup**: `on_train_begin`, `on_train_end`
+  - **Epoch Progress**: `on_epoch_begin`, `on_epoch_end`
+  - **Batch Progress**: `on_batch_begin`, `on_batch_end`
+  - **Validation**: `on_test_begin`, `on_test_end`
+  - **Prediction**: `on_predict_begin`, `on_predict_end`
+
+  The `logs` dictionary passed to these methods contains metrics like:
+  - `loss`: Training loss
+  - `val_loss`: Validation loss
+  - Any custom metrics defined in `model.compile()`
+
+  For full documentation, see [keras.callbacks.Callback](https://www.tensorflow.org/api_docs/python/tf/keras/callbacks/Callback).
+</details>
+
+## Saving Your TensorFlow Model to MLflow
+
+### Basic Model Saving
+
+If you haven't enabled autologging (which saves models automatically), you can manually save your TensorFlow model using <APILink fn="mlflow.tensorflow.log_model" />:
+
+```python
+import mlflow
+import tensorflow as tf
+from tensorflow import keras
+
+# Define model
+model = keras.Sequential(
+    [
+        keras.Input([28, 28, 3]),
+        keras.layers.Conv2D(8, 2),
+        keras.layers.MaxPool2D(2),
+        keras.layers.Flatten(),
+        keras.layers.Dense(2),
+        keras.layers.Softmax(),
+    ]
+)
+
+# Train model (code omitted for brevity)
+
+# Log the model to MLflow
+with mlflow.start_run() as run:
+    mlflow.tensorflow.log_model(model, name="model")
+
+# Later, load the model for inference
+loaded_model = mlflow.tensorflow.load_model(f"runs:/{run.info.run_id}/model")
+predictions = loaded_model.predict(tf.random.uniform([1, 28, 28, 3]))
+```
+
+<details>
+  <summary>Understanding MLflow Model Saving</summary>
+
+  When you save a TensorFlow model with MLflow:
+
+  1. **Format Conversion**: The model is converted to a generic MLflow `pyfunc` model
```

Comment:
Do we want to mention this? Cuz users can still load it back with tensorflow.load_model as in above example